test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -528,15 +528,25 @@ private synchronized NodeAndClient getRandomNodeAndClient(Predicate<NodeAndClien
      * stop any of the running nodes.
      */
     public void ensureAtLeastNumDataNodes(int n) {
+        ensureAtLeastNumDataNodes(n, Settings.EMPTY);
+    }
+
+    /**
+     * Ensures that at least <code>n</code> data nodes are present in the cluster.
+     * if more nodes than <code>n</code> are present this method will not
+     * stop any of the running nodes.  Each newly started node will use the given
+     * input settings.
+     */
+    public void ensureAtLeastNumDataNodes(final int n, final Settings settings) {
```

Comment:
this method can go away now. The settings here apply only to the nodes that are started in addition to the existing nodes that exist.